packages/neuron-wallet/src/services/tx/transaction-generator.ts

```diff
@@ -130,6 +165,107 @@ export class TransactionGenerator {
       witnesses: [],
     }
   }
+
+  public static generateDepositTx = async (
+    lockHashes: string[],
+    capacity: string,
+    receiveAddress: string,
+    changeAddress: string,
+    fee: string = '0',
+    feeRate: string = '0'
+  ): Promise<TransactionWithoutHash> => {
+    const { codeHash, outPoint, hashType } = await LockUtils.systemScript()
+    const blake160: string = LockUtils.addressToBlake160(receiveAddress)
+    const daoScriptInfo = await DaoUtils.daoScript()
+
+    const feeInt = BigInt(fee)
+    const feeRateInt = BigInt(feeRate)
+    const mode = new FeeMode(feeRateInt)
+
+    const capacityInt: bigint = BigInt(capacity)
+
+    const sizeWithoutInputs: number = TransactionGenerator.txSerializedSizeInBlockWithoutInputsForDeposit()
+    const feeWithoutInputs: bigint = TransactionGenerator.txFee(sizeWithoutInputs, feeRateInt)
+
+    const output: Cell = {
+      capacity: capacity,
+      lock: {
+        codeHash,
+        hashType,
+        args: blake160,
+      },
+      type: {
+        codeHash: daoScriptInfo.codeHash,
+        hashType: daoScriptInfo.hashType,
+        args: '0x',
+      },
+      data: '0x0000000000000000',
+      daoData: '0x0000000000000000',
```

Comment:
Where is the daoData used?

Reply:
In db, and it's the foundation to check it's a `deposit` or a `withdraw` cell.